spec.html

```diff
@@ -34781,80 +34733,51 @@ <h1>%TypedArray%.prototype.reverse ( )</h1>
-            1. Let _target_ be the *this* value.
-            1. Perform ? RequireInternalSlot(_target_, [[TypedArrayName]]).
-            1. Assert: _target_ has a [[ViewedArrayBuffer]] internal slot.
-            1. Let _targetOffset_ be ? ToIntegerOrInfinity(_offset_).
-            1. If _targetOffset_ &lt; 0, throw a *RangeError* exception.
-            1. Let _targetBuffer_ be _target_.[[ViewedArrayBuffer]].
-            1. If IsDetachedBuffer(_targetBuffer_) is *true*, throw a *TypeError* exception.
-            1. Let _targetLength_ be _target_.[[ArrayLength]].
-            1. Let _targetName_ be the String value of _target_.[[TypedArrayName]].
-            1. Let _targetElementSize_ be the Element Size value specified in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for _targetName_.
-            1. Let _targetType_ be the Element Type value in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for _targetName_.
-            1. Let _targetByteOffset_ be _target_.[[ByteOffset]].
-            1. Let _src_ be ? ToObject(_array_).
-            1. Let _srcLength_ be ? LengthOfArrayLike(_src_).
-            1. If _targetOffset_ is +&infin;, throw a *RangeError* exception.
-            1. If _srcLength_ + _targetOffset_ &gt; _targetLength_, throw a *RangeError* exception.
-            1. Let _targetByteIndex_ be _targetOffset_ &times; _targetElementSize_ + _targetByteOffset_.
-            1. Let _k_ be 0.
-            1. Let _limit_ be _targetByteIndex_ + _targetElementSize_ &times; _srcLength_.
-            1. Repeat, while _targetByteIndex_ &lt; _limit_,
-              1. Let _Pk_ be ! ToString(𝔽(_k_)).
-              1. Let _value_ be ? Get(_src_, _Pk_).
-              1. If _target_.[[ContentType]] is ~BigInt~, set _value_ to ? ToBigInt(_value_).
-              1. Otherwise, set _value_ to ? ToNumber(_value_).
-              1. If IsDetachedBuffer(_targetBuffer_) is *true*, throw a *TypeError* exception.
-              1. Perform SetValueInBuffer(_targetBuffer_, _targetByteIndex_, _targetType_, _value_, *true*, ~Unordered~).
-              1. Set _k_ to _k_ + 1.
-              1. Set _targetByteIndex_ to _targetByteIndex_ + _targetElementSize_.
-            1. Return *undefined*.
-          </emu-alg>
-        </emu-clause>
-
-        <emu-clause id="sec-%typedarray%.prototype.set-typedarray-offset">
-          <h1>%TypedArray%.prototype.set ( _typedArray_ [ , _offset_ ] )</h1>
-          <p>Sets multiple values in this _TypedArray_, reading the values from the _typedArray_ argument object. The optional _offset_ value indicates the first element index in this _TypedArray_ where values are written. If omitted, it is assumed to be 0.</p>
+        <emu-clause id="sec-settypedarrayfromtypedarray" aoid="SetTypedArrayFromTypedArray" oldids="sec-%typedarray%.prototype.set-typedarray-offset">
+          <h1>SetTypedArrayFromTypedArray ( _target_, _targetOffset_, _source_ )</h1>
```

Comment:
Nit: I generally expect "source" to come before "target". Is there a reason not to do that here?

Reply:
Hm. The spec has a few examples of target before source:
- `CopyDataProperties ( _target_, _source_, _excludedItems_ )`
- `Object.assign ( _target_, ..._sources_ )`
- `FlattenIntoArray ( _target_, _source_, _sourceLen_, ... )`
- `CopyDataBlockBytes ( _toBlock_, _toIndex_, _fromBlock_, _fromIndex_, ... )`

I'm not finding any examples of source before target.

I suppose I did target before source because that's the order they're given in the name of the operation. Ditto the InitializeTypedArrayFromX operations.

Also, with a set of related operations, it makes a bit more sense to me to have the fixed/common part of the parameter list at the left end. That might be an influence of object-oriented style; e.g., if you think of SetTypedArrayFromX as (meta) methods on the TypedArray 'class', then it's natural for the TypedArray parameter (the target) to be first.

Reply:
> I'm not finding any examples of source before target.

I was mostly thinking of the greater world: `cp` copies from its first argument to its second, e.g., and in natural language I'd usually say "copy from this [source] to that [target]".

But I hadn't realized the spec was already consistently doing `target`, `source`. I'm fine with sticking with that convention.